lib/identity_cache/query_api.rb

```diff
@@ -259,11 +259,16 @@ def prefetch_associations(associations, records)
             if details[:embed] == true
               raise ArgumentError.new("Embedded belongs_to associations do not support prefetching yet.")
             else
+              reflection = details[:association_reflection]
+              if reflection.options[:polymorphic]
+                raise ArgumentError.new("Polymorphic belongs_to associations do not support prefetching yet.")
```

Comment:
This is for simplicity, since when dealing with multiple records, we might have a mix of types to fetch.  It could be implemented by grouping by type to fetch them, but I'm not sure if we want to support that complexity at the moment.